src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -913,6 +922,114 @@ public Page search(String q) throws Exception {
     }
 
     /**
+     * Get JSON from A Jenkins endpoint.
+     * @param path The endpoint URL.
+     * @return The JSON.
+     */
+    public JSONWebResponse getJSON(@Nonnull String path) throws IOException {
+        assert !path.startsWith("/");
+
+        JenkinsRule.WebClient webClient = createWebClient();
+        Page runsPage = null;
+        try {
+            runsPage = webClient.goTo(path, "application/json");
+        } catch (SAXException e) {
+            // goTo shouldn't be throwing a SAXException for JSON.
+            throw new IllegalStateException("Unexpected SAXException.", e);
+        }
+        WebResponse webResponse = runsPage.getWebResponse();
+
+        return new JSONWebResponse(webResponse);
+    }
+
+    /**
+     * POST a JSON payload to a URL on the underlying Jenkins instance.
+     * @param path The url path on Jenkins.
+     * @param json An object that produces a JSON string from it's {@code toString} method.
+     * @return A JSON response.
+     * @throws IOException
+     * @throws SAXException
+     */
+    public JSONWebResponse postJSON(@Nonnull String path, @Nonnull Object json) throws IOException, SAXException {
+        assert !path.startsWith("/");
+
+        URL postUrl = new URL(getURL().toExternalForm() + path);
+        HttpURLConnection conn = (HttpURLConnection) postUrl.openConnection();
+
+        conn.setDoOutput(true);
+        long startTime = System.currentTimeMillis();
+
+        try {
+            conn.setRequestMethod("POST");
+            conn.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+
+            NameValuePair crumb = getCrumbHeaderNVP();
+            conn.setRequestProperty(crumb.getName(), crumb.getValue());
+
+            byte[] content = json.toString().getBytes(UTF8);
+            conn.setRequestProperty("Content-Length", String.valueOf(content.length));
+            final OutputStream os = conn.getOutputStream();
+            try {
+                os.write(content);
+                os.flush();
+            } finally {
+                os.close();
```

Comment:
:-1:, simpler and better as written (and when `-source 7` can use a `try`-with-resources block).